### PR TITLE
Update README.MD

### PR DESCRIPTION
**Pull request by joaquinstan**
Improved README.MD:
 - Added section about custom nginx configuration
 - Mentioned possible issues with login if domain is not whitelisted
 - Explained meaning of variables
 - Improved general structure

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -1,16 +1,57 @@
...
-Use the docker image generated from this project to setup a reverse proxy for your organization's public portal using nginx.
+This repository shows the necessary nginx configuration for setting up an organization's public portal using a vanity domain. It also provides a docker image for running this configuration.
 
-## Usage
+## Whitelisting your domain
 
-Get the image from docker hub:
+To allow the Exchange public portal to work with your own domain, it first needs to be whitelisted. To do so:
+
+1. Get an access token from an administrator user of your organization in the Anypoint Platform:
```
Comment (nicolasgnr) [inline README.MD]: `... from a user with Organization Administrators role in the master organization`?

File: README.MD [hunk 2]
```diff
@@ -1,16 +1,57 @@
...
+
+```
+curl https://anypoint.mulesoft.com/exchange/api/v1/organizations/YOUR_ORGANIZATION_DOMAIN/portal/domain -XPUT -d "your.custom.domain" -H "content-type: text/plain" -H "authorization: bearer ACCESS_TOKEN"
+```
+
+Keep in mind that if this step is not followed, you wont be able to log in to the portal or perform any actions that need authentication.
+
+## Setting up your own nginx configuration
+
+If your organization already has a running nginx configuration, you must include the necessary rules found [here](https://github.com/mulesoft-labs/anypoint-exchange-public-portal-proxy-image/blob/master/config/nginx.conf.template#L11-L30) replacing the following variables with what may apply:
```
Comment (nicolasgnr) [inline README.MD]: I would just mention that is mandatory to run the server using `HTTPS`

File: README.MD [hunk 3]
```diff
@@ -1,16 +1,57 @@
...
+# Exchange public portal vanity domain setup
 
-Use the docker image generated from this project to setup a reverse proxy for your organization's public portal using nginx.
+This repository shows the necessary nginx configuration for setting up an organization's public portal using a vanity domain. It also provides a docker image for running this configuration.
 
-## Usage
+## Whitelisting your domain
 
-Get the image from docker hub:
+To allow the Exchange public portal to work with your own domain, it first needs to be whitelisted. To do so:
```
Comment (mcampo) [inline README.MD]:
```suggestion
Your domain needs to be whitelisted in Exchange for your public portal to work properly. To do so:
```

File: README.MD [hunk 4]
```diff
@@ -1,16 +1,57 @@
-# Exchange Public Portals Reverse Proxy Image
+# Exchange public portal vanity domain setup
 
-Use the docker image generated from this project to setup a reverse proxy for your organization's public portal using nginx.
+This repository shows the necessary nginx configuration for setting up an organization's public portal using a vanity domain. It also provides a docker image for running this configuration.
```
Comment (mcampo) [inline README.MD]:
```suggestion
This repository contains instructions and the necessary nginx configuration for setting up an organization's public portal using a vanity domain. It also provides a docker image for running this configuration.
```

File: README.MD [hunk 5]
```diff
@@ -1,16 +1,57 @@
...
+1. Get an access token from an administrator user of your organization in the Anypoint Platform:
+
+```
+ACCESS_TOKEN=$(curl https://anypoint.mulesoft.com/accounts/login -XPOST -d "username=YOUR_USERNAME&password=YOUR_PASSWORD" | jq -r ".access_token")
+```
+
+2. Use that token to set your custom domain in your organization:
+
+```
+curl https://anypoint.mulesoft.com/exchange/api/v1/organizations/YOUR_ORGANIZATION_DOMAIN/portal/domain -XPUT -d "your.custom.domain" -H "content-type: text/plain" -H "authorization: bearer ACCESS_TOKEN"
```
Comment (mcampo) [inline README.MD]:
```suggestion
curl https://anypoint.mulesoft.com/exchange/api/v1/organizations/YOUR_ORGANIZATION_DOMAIN/portal/domain -XPUT -d "your.custom.domain" -H "content-type: text/plain" -H "authorization: bearer $ACCESS_TOKEN"
```

File: README.MD [hunk 6]
```diff
@@ -1,16 +1,57 @@
...
+
+
+## Running the docker image
+
+For a minimal working setup, there is an nginx based docker image available.
+
+1. Get the image from docker hub:
 
 ```
 docker pull joaquinstankus/anypoint-exchange-public-portal-proxy:latest
```
Comment (mcampo) [inline README.MD]: let's find out who's the owner of https://hub.docker.com/u/mulesoft/ and publish the image there

File: README.MD [hunk 7]
```diff
@@ -1,16 +1,57 @@
...
-Get the image from docker hub:
+To allow the Exchange public portal to work with your own domain, it first needs to be whitelisted. To do so:
+
+1. Get an access token from an administrator user of your organization in the Anypoint Platform:
+
+```
+ACCESS_TOKEN=$(curl https://anypoint.mulesoft.com/accounts/login -XPOST -d "username=YOUR_USERNAME&password=YOUR_PASSWORD" | jq -r ".access_token")
+```
+
+2. Use that token to set your custom domain in your organization:
```
Comment (mcampo) [inline README.MD]:
```suggestion
2. Use that token to whitelist your custom domain:
```

File: README.MD (hunk 6 above)
Comment (nicolasgnr) [reply README.MD]: +1

**Pull request merged**